### DISCUSSION TITLE
Allow multiple entrances and exits on mazes.

### DISCUSSION THREAD

**Discussion by ocalhoun6** [OpenRCT2/OpenRCT2#17645] [Feature requests]
What it says on the tin, basically.

There doesn't seem to be any fundamental reason why a maze couldn't have multiple exits, or even multiple entrances, as long as it has at least one of each. And it could make for some pretty interesting layouts. Or interesting things like having one (easy) exit lead just back to normal path, but having the other (difficult) exit lead to a free balloon stall as a 'prize' for completing the harder part of the maze, or having an 'emergency exit' from a difficult maze, etc. Or, I don't know, maybe a one-tile maze with three entrances and one exit, if somebody _really_ wants to min-max the heck out of their maze. Might also be fun for some players to toy with the maze pathfinding, allowing them to make a layout with multiple exits along different paths and see which exits the peeps find most easily.

Just seems like a feature that could make mazes a bit more fun to build, and -- _hopefully_ -- it wouldn't be very difficult to implement. It may be that once you allow the extra entrances/exits to be placed, it just works like normal ... or maybe it would be a huge headache. But I figure it's worth thinking about?

**Comment by jjpb44** [comment]
I think that the pathfinding in the maze is not a problem since the (current) algorithm does not take exits into account.
